pytorch_lightning/callbacks/progress/tqdm_progress.py

```diff
@@ -381,7 +381,9 @@ def convert_inf(x: Optional[Union[int, float]]) -> Optional[Union[int, float]]:
     return x
 
 
-def _update_n(bar: _tqdm, value: int) -> None:
-    if not bar.disable:
-        bar.n = value
-        bar.refresh()
+def _update_n(bar: _tqdm, current: int, refresh_rate: int) -> None:
+    total = bar.total
+    leftover = current % refresh_rate
+    advance = leftover if (current == total and leftover != 0) else refresh_rate
```

Comment:
Can you explain the fix? Why does setting the current value directly not work?

Reply:
because it needs to update some internal states like `last_print_t`, `last_print_t` which are used to determine the time and iterations/sec.

also `n` here should be the incremental value inside `update()` just like Rich progress bar.